### PR TITLE
[backport] [aws] Remove hardcoded event.dataset field and use ECS field instead

### PR DESCRIPTION
**Pull request by zmoog**
## Proposed commit message

Backport https://github.com/elastic/integrations/pull/8811 to 8.8.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

### CODE REVIEW

File: packages/aws/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.51.3"
+  changes:
+    - description: Remove hardcoded event.dataset field and use ecs instead.
+      type: bugfix
+      link: https://github.com/elastic/integrations/pull/8811
```
Comment (kaiyan-sheng) [inline packages/aws/changelog.yml:6]: Should the PR link be #8888 instead?

File: packages/aws/data_stream/cloudwatch_metrics/fields/base-fields.yml [hunk 2]
```diff
@@ -13,8 +13,4 @@
 - name: event.module
   type: constant_keyword
   description: Event module
-  value: aws
-- name: event.dataset
-  type: constant_keyword
-  description: Event dataset
-  value: aws.cloudwatch_metrics
+  value: aws
```
Comment (kaiyan-sheng) [inline packages/aws/data_stream/cloudwatch_metrics/fields/base-fields.yml:16]: need an extra line here?

Comment (zmoog) [reply packages/aws/data_stream/cloudwatch_metrics/fields/base-fields.yml:16]: Yep, adding one.

**Pull request merged**